sources/packages/backend/workflow/test/test-utils/mock/mock.utils.ts

```diff
@@ -0,0 +1,173 @@
+import { PublishMessageRequest, ZeebeJob } from "zeebe-node";
+import {
+  JOB_COMPLETED_RESULT_SUFFIX,
+  JOB_MESSAGE_RESULT_SUFFIX,
+  MOCKS_SEPARATOR,
+  PARENT_SUBPROCESSES_VARIABLE,
+  SERVICE_TASK_ID_SEPARATOR,
+} from "../constants/mock-constants";
+import {
+  WorkflowServiceTasks,
+  WorkflowSubprocesses,
+} from "../constants/workflow-variables-constants";
+
+/**
+ * Regex for replace all.
+ */
+const SERVICE_TASK_ID_SEPARATOR_REGEX = new RegExp(
+  SERVICE_TASK_ID_SEPARATOR,
+  "g",
+);
+
+/**
+ * Get the mock identifier for a completed job to be sent to the workflow,
+ * for instance, create_supporting_users_for_parents_task_result, where:
+ * - `create_supporting_users_for_parents_task`: service task id;
+ * - `result`: suffix that identifies the job completed object.
+ * @param serviceTaskId service task id that will have the job completed returned.
+ * @returns mock identifier for a completed job to be sent to the workflow.
+ */
+export function getServiceTaskResultMockId(serviceTaskId: string) {
+  return `${getNormalizedServiceTaskId(
+    serviceTaskId,
+  )}${MOCKS_SEPARATOR}${JOB_COMPLETED_RESULT_SUFFIX}`;
+}
+
+/**
+ * Get the mock identifier for a message to be sent to the workflow,
+ * for instance, create_supporting_users_for_parents_task_messageResult, where:
+ * - `create_supporting_users_for_parents_task`: service task id;
+ * - `messageResult`: suffix that identifies a message payload to be sent to the
+ * workflow.
+ * @param serviceTaskId service task id that will need to publish the message.
+ * @returns mock identifier for a message to be sent to the workflow.
+ */
+export function getPublishMessageResultMockId(serviceTaskId: string) {
+  return `${getNormalizedServiceTaskId(
+    serviceTaskId,
+  )}${MOCKS_SEPARATOR}${JOB_MESSAGE_RESULT_SUFFIX}`;
+}
+
+/**
+ * Convert the service task id, usually declared as 'service-task-id' to the expected
+ * Camunda variable name like service_task_id.
+ * The usual variables along the workflow are following the camelCase pattern but service
+ * task ids are actually following the kebab-case pattern, which is not a problem in general.
+ * Based on it, while trying to use the service task ids as variables names we can
+ * respect the Camunda recommendations (link below) and just convert the kebab-case to
+ * snake_case pattern.
+ * @see https://docs.camunda.io/docs/components/concepts/variables/#variable-names
+ * @param serviceTaskId workflow service task id.
+ * @returns service task id, usually declared as 'service-task-id' to the expected
+ * Camunda variable name like service_task_id.
+ */
+function getNormalizedServiceTaskId(serviceTaskId: string) {
+  return serviceTaskId.replace(
```

Comment:
I would assume that if we will be changing the way that we are using the IDs we would be changing everywhere, not limited to only tasks, and also renaming the subprocesses calls.
We can do it but I do not believe that it would be the scope of this PR.